pkg/operator/operator.go

```diff
@@ -225,3 +236,8 @@ func kubeDNSIP(ctx context.Context, kubernetesInterface kubernetes.Interface) (n
 	}
 	return kubeDNSIP, nil
 }
+
+func setDurationAndExpiry(provider *stscreds.AssumeRoleProvider, ctx context.Context) {
```

Comment:
Should we set the duration for ever credential provider if we are going to have that be a passed through option? It seems odd that we have a parameter named `credsDuration` but it only applies to the assume role provider creds

Reply:
I looked into this a little bit. This setting seems to be a provider specific settings. Some providers have this configuration (WebIdentityRoleProvider) while others do not (EC2RoleProvider). It is possible to instantiate a new session using `session.NewSessionWithOptions` and pass in an options struct. This struct does have a field for setting credentials duration but the field is called `AssumeRoleDuration` and the docs state that this field is only used when the sdk shared config provides a role to assume. 

As far as I can tell, there is no duration field that is generally applicable to all providers. Would renaming `credsDuration` to `assumeRoleDuration` make more sense?

Reply:
Agree the duration is an assume role concept. Even the "WebIdentityRoleProvider" calls "AssumeRoleWithWebIdentity"

> Would renaming credsDuration to assumeRoleDuration make more sense?

✔️

Reply:
Fixed in next revision.